sway-ir/src/optimize/arg_demotion.rs

```diff
@@ -123,21 +123,19 @@ fn demote_fn_signature(context: &mut Context, function: &Function, arg_idcs: &[(
         .collect::<Vec<_>>();
 
     // For each of the old args, which have had their types changed, insert a `load` instruction.
-    let arg_val_pairs = old_arg_vals
```

Comment:
MUCH nicer with the new API all the places in this file. No need for `rev()` traversal and manual direct manipulation of the `instructions`.